crates/telio-starcast/src/transport.rs

```diff
@@ -118,6 +144,9 @@ impl Transport {
                 recv_buffer: vec![0; MAX_PACKET_SIZE],
                 nat: StarcastNat::new(IPV4_STARCAST_ADDRESS, IPV6_STARCAST_ADDRESS),
                 peers: Vec::new(),
+                socket_pool,
+                meshnet_ip,
+                exponential_backoff,
```

Comment:
Is there any way in which we may lose the socket? If not maybe we could drop all of this after getting one? (E.g. Making an enum which either keeps socket or these guys?)

Reply:
Seems like an overcomplication, since these types don't really take up large amounts of memory or anything, unless I'm missing something.

Reply:
I don't insist, but I guess that would be a cleaner design, instead of matching on `Some(socket)` or `None`, you would match on `SocketPreparation(socket_pool, ..)` or `SocketReady(socket)` - I think it would be easier to catch some kinds of errors (like the one below) and the `State` wouldn't be polluted with some fields which are actually used only once for this socket creation.

Reply:
I'd really rather leave it as it is, if it's not causing any issues.

Reply:
Sure